File: src/main/java/com/bdev/hogwarts_api/data/dto/student/StudentAttendance.java
package com.bdev.hogwarts_api.data.dto.student;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Builder;
import lombok.Getter;
import lombok.NonNull;

@Builder
@Getter
public class StudentAttendance {
    private Long id;
    @NonNull private Long studentId;
    @NonNull private StudentAttendanceType type;
    @NonNull private Long time;

    @JsonCreator
    public static StudentAttendance create(
       @JsonProperty("id") Long id,
       @JsonProperty("studentId") Long studentId,
       @JsonProperty("type") StudentAttendanceType type,
       @JsonProperty("time") Long time
    ) {
        return StudentAttendance
                .builder()
                .id(id)
                .studentId(studentId)
                .type(type)
                .time(time)
                .build();
    }
}

File: src/main/java/com/bdev/hogwarts_api/data/dto/student/StudentAttendance.kt
package com.bdev.hogwarts_api.data.dto.student

import com.bdev.hogwarts_api.data.dto.group.GroupType
import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonProperty

data class StudentAttendanceId @JsonCreator constructor(
        @JsonProperty("studentLogin") val studentLogin: String,
        @JsonProperty("startTime") val startTime: Long,
        @JsonProperty("finishTime") val finishTime: Long
)

data class StudentAttendance @JsonCreator constructor(
        @JsonProperty("studentLogin") val studentLogin: String,
        @JsonProperty("type") val type: StudentAttendanceType,
        @JsonProperty("groupType") val groupType: GroupType,
        @JsonProperty("studentsInGroup") val studentsInGroup: Int,
        @JsonProperty("startTime") val startTime: Long,
        @JsonProperty("finishTime") val finishTime: Long,
        @JsonProperty("ignoreSingleStudentPricing") val ignoreSingleStudentPricing: Boolean
)
